jbmc/src/janalyzer/Makefile

```diff
@@ -38,6 +39,11 @@ ifneq ($(wildcard ../jsil/Makefile),)
   CP_CXXFLAGS += -DHAVE_JSIL
 endif
 
+ifneq ($(CUDD),)
```

Comment:
Rather than having to sprinkle the same if block + compile config throughout lots of different makefiles, would it be possible to stick this if block into a `cudd.inc` file somewhere, then simply just unconditionally `include path/to/cudd.inc` everywhere instead?

Reply:
In fact... could the `cudd.inc` file also do the `OBJ += ...` line as well?

Reply:
I'll try to make that work.

Reply:
I have now moved most of the rules to `src/common`, shrinking this PR quite a bit. Just the `OBJ` parts I have left in place as we only very rarely, and also not really uniformly, need those.